File: practicas/4/RubioAngel/Hilos.py
import threading
import time

# Función que ejecutará el hilo
def contar(nombre, n):
    with open("resultado.txt", "a") as f:
        for i in range(1, n+1):
            mensaje = f"{nombre} cuenta: {i}\n"
            print(mensaje, end="")
            f.write(mensaje)
            time.sleep(0.5)

# Crear dos hilos
hilo1 = threading.Thread(target=contar, args=("Hilo 1", 5))
hilo2 = threading.Thread(target=contar, args=("Hilo 2", 5))

# Iniciar los hilos
hilo1.start()
hilo2.start()

# Esperar a que los hilos terminen
hilo1.join()
hilo2.join()
#Agregamos solo un comentario para que se visualice
print("Comentario")
print("Tarea finalizada. Revisa 'resultado.txt'.")

File: practicas/4/TorresSamuel/Hilos.java
class Hilo1 extends Thread {
    @Override
    public void run() {
        for (int i = 1; i <= 5; i++) {
            try {
                Thread.sleep(1000);
            } catch (InterruptedException e) {
                System.out.println(e);
            }
            System.out.println("Número del hilo 1: " + i);
        }
    }
}

class Hilo2 extends Thread {
    @Override
    public void run() {
        for (int i = 6; i <= 10; i++) {
            try {
                Thread.sleep(1000);
            } catch (InterruptedException e) {
                System.out.println(e);
            }
            System.out.println("Número del hilo 2: " + i);
        }
    }
}

public class Hilos {
    public static void main(String[] args) {
        Hilo1 hilo1 = new Hilo1();
        Hilo2 hilo2 = new Hilo2();

        hilo1.start();
        hilo2.start();

        try {
            hilo1.join();
            hilo2.join();
        } catch (InterruptedException e) {
            System.out.println(e);
        }

        System.out.println("Este es el hilo principal.");
    }
}

